pkg/constants/constants.go

```diff
@@ -37,4 +37,8 @@ const (
 
 	WorkloadPriorityClassSource = "kueue.x-k8s.io/workloadpriorityclass"
 	PodPriorityClassSource      = "scheduling.k8s.io/priorityclass"
+
+	// PreemptionAdmissionCheckName is the name of the builtin preemption
+	// admission check controller.
+	PreemptionAdmissionCheckName = "kueue-preemption"
```

Comment:
This is used as a condition type, so should be camel cased according to: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1498-L1507

Reply:
For this builtin admission-check it is doable, however other should hold the object name of the admission check name which need to be DNS subdomain.

Reply:
It would be good o be compatible with k8s recommendation. Is there a way? Maybe we could have a DNS name & condition type name per admission check declared at the cluster queue level? wdyt @mwielgus @trasc ?

Reply:
To follow the proposed pattern ` (dns1123SubdomainFmt/)?(qualifiedNameFmt)` we can append something like `/CheckCondition` to the admission check name, however for me it looks a bit "artificial" and require  additional
strings manipulation with no added value.

An alternative could to have a custom Condition type,  maybe we can look at this in a followup.

Reply:
> To follow the proposed pattern  (dns1123SubdomainFmt/)?(qualifiedNameFmt) we can append something like /CheckCondition to the admission check name, however for me it looks a bit "artificial" and require additional
strings manipulation with no added value.

Agree, I'm not a big fun of artificial constructs. IMO it's fine as is, if there is no good option otherwise. Is there a way providing a mapping between the names, or lowercasing?

> An alternative could to have a custom Condition type, maybe we can look at this in a followup.

If we plan for follow up, it would mean changing the API, and the conditions might be used by external controllers, IIUC, so it will be costly,

Reply:
Could you point out in code where the name used as a DNS domain name?

Reply:
The `Type` of the conditions is extracted from the list of checks associated with a clusterQueue, which is a list of `AdmissionChecks` identified by their names (kubernetes resource name).

Reply:
I don't think that we should create yet another mapping in clusterqueue spec. Maybe keeping the same name everywhere is not so bad idea :) ?

Reply:
sgtm